From 71fe2f219c561ff5e3cf4f0d4fa93977b3715a23 Mon Sep 17 00:00:00 2001
From: Hendrik Leppkes <h.leppkes@gmail.com>
Date: Sun, 2 Feb 2014 16:53:10 +0100
Subject: [PATCH 31/62] avformat: try harder to find a duration of the media

This helps finding a duration in files with a lot of garbage data at the
end, increasing the lookup area from 4MB to 32MB.

There should be no speed loss on files which previously worked fine, since
it'll stop looking once it found a duration.
---
 libavformat/utils.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavformat/utils.c b/libavformat/utils.c
index 5bc8259..f78e6d7 100644
--- a/libavformat/utils.c
+++ b/libavformat/utils.c
@@ -2466,7 +2466,7 @@ static void estimate_timings_from_bit_rate(AVFormatContext *ic)
 }
 
 #define DURATION_MAX_READ_SIZE 250000LL
-#define DURATION_MAX_RETRY 6
+#define DURATION_MAX_RETRY 7
 
 /* only usable for MPEG-PS streams */
 static void estimate_timings_from_pts(AVFormatContext *ic, int64_t old_offset)
-- 
2.6.3

